asv/environment.py

```diff
@@ -327,14 +327,11 @@ def get_environments(conf, env_specifiers, verbose=True):
             # Ignore requirement matrix
             requirements_iter = [dict(python=python) for python in pythons]
 
-        env_matrix_combinations = iter_env_matrix_combinations(conf.env_matrix)
-
-        if not env_matrix_combinations:
-            env_matrix_combinations = [{}]
-
         for requirements in requirements_iter:
             python = requirements.pop('python')
 
+            env_matrix_combinations = iter_env_matrix_combinations(conf.env_matrix)
```

Comment:
Putting this iterable outside of the loop was deliberate since it is invariant. Is there something I'm missing?

Reply:
Ha, of course "it's a generator".